### PR TITLE
#06 Chore Swagger-UI Documentation 

### PR DESCRIPTION
**Pull request by arthurjunior250**
**What does this PR do?**
Implements Swagger-UI Documentation of Endpoints
**Description of Task to be completed?**
Document my API with Swagger.

**How should this be manually tested?**
Clone the repo

checkout to branch ch-documentation-swagger--06
-Run `yarn install to install modules

-Run yarn start:dev to start

In your Browser

-type: http://127.0.0.1:3000/api-docs/ to be able to access the documentation
**Any background context you want to provide?**
N/A

Screenshots (if appropriate)
Screenshot from 2022-03-14 16-51-57
Screenshot from 2022-03-14 16-52-07
Screenshot from 2022-03-14 16-52-11

### CODE REVIEW

**Review by shemalucien — APPROVED** [review]
good work mn

**Review by PatrickNiyogitare28 — CHANGES_REQUESTED** [review]
Good work, but you need to add the `.nyc_output` file under `.gitignore`. We don't need to push it to GitHub

File: src/database/index.js [hunk 1]
```diff
@@ -2,16 +2,21 @@ import mongoose from 'mongoose';
...
 
-const db_url = (env == 'production') ? process.env.DB_PRODUCTION_URL : process.env.DB_DEVELOPMENT_URL
+const dev_db_url = process.env.DB_DEVELOPMENT_URL;
+const prod_db_url = process.env.DB_PRODUCTION_URL;
+const test_db_url = process.env.DB_TESTING_URL;
+const connectionUrl = env == "development" ? dev_db_url : env == "production" ? prod_db_url : test_db_url;
 
-mongoose.connect(db_url, {
+// const db_url = (env == 'production') ? process.env.DB_PRODUCTION_URL : process.env.DB_DEVELOPMENT_URL
+
```
Comment (PatrickNiyogitare28) [inline src/database/index.js:11]: Just remove this comment

File: src/routes/blog.route.js [hunk 2]
```diff
@@ -14,9 +14,14 @@ const fileFilter = (req, file, cb) => {
...
 const uploads = multer({ storage, fileFilter });
 
-router.post('/', checkAdmin, uploads.single("image"), saveBlog);
-router.put('/:id', checkAdmin, updateBlog);
-router.delete('/:id', checkAdmin, deleteBlogById);
+// router.post('/', checkAdmin, uploads.single("image"), saveBlog);
+// router.put('/:id', checkAdmin, updateBlog);
+// router.delete('/:id', checkAdmin, deleteBlogById);
+// router.get('/', getAllBlogs);
+// router.get('/:id', getblogById);
```
Comment (PatrickNiyogitare28) [inline src/routes/blog.route.js:21]: Remove the commented sources codes for the best code quality

File: src/routes/comment.route.js [hunk 3]
```diff
@@ -1,8 +1,12 @@
 import express from 'express';
 import { deleteComment, saveComment, getAllComments } from '../controllers/comment.controller';
-import { check } from "../middleware/check";
+import { checkUser } from "../middleware/check";
 const router = express.Router();
 
+// router.post("/:id", checkUser, saveComment);
+// router.get("/:id", getAllComments);
+// router.delete("/:id", checkUser, deleteComment);
+
```
Comment (PatrickNiyogitare28) [inline src/routes/comment.route.js:9]: Remove the comments here as well

File: src/routes/inquiry.route.js [hunk 4]
```diff
@@ -4,8 +4,11 @@ import { checkAdmin } from "../middleware/check";
 const router = express.Router();
 
 router.post('/', saveInquiry);
-router.get('/', checkAdmin, getAllInquiries);
-router.get('/:id', checkAdmin, getInquiryById);
-router.delete('/:id', checkAdmin, deleteInquiryById);
+// router.get('/', checkAdmin, getAllInquiries);
+// router.get('/:id', checkAdmin, getInquiryById);
+// router.delete('/:id', checkAdmin, deleteInquiryById);
```
Comment (PatrickNiyogitare28) [inline src/routes/inquiry.route.js:9]: Take care of the comments

File: src/routes/newsletter.route.js [hunk 5]
```diff
@@ -3,7 +3,10 @@ import { newsLetterEmail, deleteNewsById, getAllSubscribers } from '../controlle
 import { checkAdmin } from "../middleware/check";
 const router = express.Router();
 
+// router.post('/', newsLetterEmail);
+// router.delete('/:id', checkAdmin, deleteNewsById);
+// router.get('/', checkAdmin, getAllSubscribers);
```
Comment (PatrickNiyogitare28) [inline src/routes/newsletter.route.js:8]: Take care of the comments

**Review by PatrickNiyogitare28 — CHANGES_REQUESTED** [review]
Take care of the comments. you can uncomment the codes or remove the comments

**Pull request merged**